cmake/CopyTarget.cmake

```diff
@@ -0,0 +1,145 @@
+# Provides the copy_target() function, which is used to make duplicate MPI and OpenMP versions of targets.
+
+# copies the value of a property from one target to another.  Does nothing if the property is not defined on the SOURCE.
+function(copy_property PROPERTY SOURCE DESTINATION)
+	get_property(PROP_IS_DEFINED TARGET ${SOURCE} PROPERTY ${PROPERTY} SET)
```

Comment:
Since you put `SET` here, does that mean in the future calls the copied property is alway a boolean?

Reply:
No, this gets if the property ${PROPERTY} is defined for the current target.

Reply:
In the documentation it says:
`If the SET option is given the variable is set to a boolean value indicating whether the property has been set.` What `variable` is being talked about here? If it's not ${PROP_IS_DEFINED}, what actually does `SET` do?